sort_image_files.sh

```diff
@@ -0,0 +1,223 @@
+
+    date_for_date_change="${exif_date//:/}"
+    date_for_date_change="${date_for_date_change// /}"
+    # Trim last 2 chars to remove SS so in format: [[CC]YY]MMDDhhmm[.SS]
+    date_for_date_change="${date_for_date_change%??}.${date_for_date_change: -2}"
+
+    # Prepare Filename Change that includes Date
+    # Use EXIF photo-taken date, EXIF:DateTimeOriginal, change format & use in filename.
+    # 1. Replace last 3 chars, e.g., :17	With: nothing
+    # 2. Replace ALL : 	With: -
+    # 3. Replace ALL spaces  With: _
+    
+    date_for_filename_change="${exif_date/${exif_date: -3}}"
+    date_for_filename_change="${date_for_filename_change//:/-}"
+    date_for_filename_change="${date_for_filename_change// /_}"
+  
+    # Build onto new_file_name. Concat to front - year & month.
+    # Use: ${string:position:length}   On: 2015:09:02 07:09:03
+    year="${exif_date:0:4}"
+    month="${exif_date:5:2}"
+  else # Use modify date when no exif date.
+    # Change filesystem date to modify_date, date:modify for date change.
+    # Given Format:  2018-10-09T18:42:41+00:00
+    # Wanted Syntax: [[CC]YY]MMDDhhmm[.SS] Wanted Format: 202002031806
+    # 1. Trim last 9 chars.
+    # 2. Remove all -.
+    # 3. Remove all T.
+    # 4. Remove all :.
+    date_for_date_change="${modify_date::-9}" # Result: 2020-02-08T16:02
+    date_for_date_change="${date_for_date_change//-/}" # Result: 20200208T16:02
+    date_for_date_change="${date_for_date_change//T/}" # Result: 2020020816:02
+    date_for_date_change="${date_for_date_change//:/}" # Result: 202002081602
+    
+    # Prepare Filename Change that includes Date. Use modify_date, change format
+    # & use in filename.
+    # Given Format:  2018-10-09T18:42:41+00:00
+    # Wanted Format: 2015-09-02_07-09_0060.jpg
+    # 1. Replace last 3 chars, e.g., :17	With: nothing
+    # 2. Replace ALL : 	With: -
+    # 3. Replace ALL spaces  With: _
+    date_for_filename_change="${modify_date::-9}" # Result: 2018-10-09T18:42
+    date_for_filename_change="${date_for_filename_change//T/-}" # Result: 2018-10-09_18:42
+    date_for_filename_change="${date_for_filename_change//:/_}" # Result: 2018-10-09_18_42
+  
+    # Build onto new_file_name. Concat to front - year & month.
+    # Use: ${string:position:length}   On: 2018-10-09T18:42:41+00:00
+    year="${modify_date:0:4}"
+    month="${modify_date:5:2}"
+  fi
+
+  just_path=$(dirname "${a_file_name}")
+  just_filename=$(basename "${a_file_name}") # For path to move files into subdirectories
+
+  # 4 Cases for Options to Add Day & Rename Files:
+  # Case 1: Add Day AND Rename files.
+  # Case 2: Add Day. No Rename files.
+  # Case 3: No add Day. Rename files.
+  # Case 4: No add Day. No Rename files.
+
+  day="${date_for_date_change:6:2}"
+  path_with_year_month_day="${just_path}/${year}/${month}/${day}"
+  path_with_year_month="${just_path}/${year}/${month}"
+
+  if [[ "${clean_day_subdir_also}" == 'y' ]] && [[ "${clean_rename_files_also}" == 'y' ]]; then
```

Comment:
Since you're using the `[[ ]]` form of the test, you can use the `&&` operator inside the double square brackets and shorten to

```shell
if [[ "${clean_day_subdir_also}" == 'y' && "${clean_rename_files_also}" == 'y' ]]; then
```

Reply:
@kcharter Removed extra unneeded ]] [[ as above.

Reply:
@kimlew Looks good.